packages/myst-directives/src/figure.ts

```diff
@@ -47,20 +47,51 @@ export const figureDirective: DirectiveSpec = {
     //   type: ParseTypesEnum.string,
     //   // class_option: list of strings?
     // },
+    'remove-input': {
+      type: ParseTypesEnum.boolean,
+    },
+    'remove-output': {
+      type: ParseTypesEnum.boolean,
+    },
+    placeholder: {
+      type: ParseTypesEnum.string,
+    },
   },
   body: {
     type: ParseTypesEnum.parsed,
   },
   run(data: DirectiveData): GenericNode[] {
-    const img: Image = {
-      type: 'image',
-      url: data.arg as string,
-      alt: data.options?.alt as string,
-      width: data.options?.width as string,
-      height: data.options?.height as string,
-      align: data.options?.align as Image['align'],
-    };
-    const children: GenericNode[] = [img];
+    const url = data.arg as string;
+    const children: GenericNode[] = [];
+    // If figure arg starts with #, use embed node. Otherwise use image node.
+    if (url.startsWith('#')) {
```

Comment:
Let's leave this logic in the image transform so it's in one place. The `remove-` options can just be tossed on the image nodes.